### PR TITLE
Node FIXED & Updated (and install.sh corrected)

### PR DESCRIPTION
**Pull request by thedamian**
Fixes issue: https://github.com/skycocker/chromebrew/issues/155 and https://github.com/skycocker/chromebrew/issues/143 and https://github.com/skycocker/chromebrew/issues/138

Node is Fixed (sorry for the delay, ready below as to why it took so long if interested)
ALSO the install.sh ln of the gcc to cc (to fix some of the errors out there)

-- What too so long --
I went to the dark side (first moved to crew and then full ubuntu on my chromebook pixel) 

But I'm back... and I noticed NODE and NODE_STABLE wasn't installing and my install.sh contribution to make some app builds work wasn't in sudo (which was needed) 

the node fix came from stealing of the LinuxBrew formula. 
(Always thank the giants who's shoulders you stand on while urinating really far)

ALSO changed the name of "NODE_STABLE" to "NODE_CURRENT" to follow the naming convention of node (NODE!: Stop changing how you name everything!)

### CODE REVIEW

**Comment by thedamian** [comment]
to use these files yourself (to test or to fix the issues) simply copy the *.rb files to /usr/local/crew/packages/ and you'll be able to do a 

crew install node_current 

and you'll have node with NPM installed. (Now a days with that you can almost do ANYTHING!)

**Comment by thedamian** [comment]
Mono package added. .Net is coming people! 
And the permissions to make a lot of packages easier to install

File: packages/dart.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+require 'package'
+
+class Dart < Package
+  version '4.4.0.148'
+
+    source_url ({
+    i686: "https://www.dropbox.com/s/d1omw087ilkh5oq/jdk1.8.0_25_i686.tar.gz?dl=0",
+    x86_64: "https://www.dropbox.com/s/ykw2jt797cpghfy/jdk1.8.0_65_x86_64.tar.gz?dl=0"
+  })
```
Comment (skycocker) [inline packages/dart.rb]: @thedamian I'm always trying not to complain about indentation, but this is so visible 😀

**Comment by skycocker** [comment]
@thedamian what about users who already have node_stable installed?

**Comment by thedamian** [comment]
Skycocker. Did I catch you

it's all fixed.

**Comment by thedamian** [comment]
Just did a powerwash of my chromebook. Went to regular user and back to developer mode (still in stable channel) and then ran MY install.sh with success and installed NODE from my version of the projects. 

everything is looking good.

I tried pull 178 (the removing of the sudo) but it didn't work. nothing got created and nothing was running (hence why I had to do a powerwash)

**Comment by skycocker** [comment]
Looks like it's fine then. Thanks for all the work!

**Pull request merged**